internals/search.py

```diff
@@ -91,12 +93,28 @@ def process_recent_reviews_query() -> list[int] | Future:
   return future_feature_ids
 
 
+@dataclasses.dataclass
+class QueryContext:
+  now: datetime.datetime
+  current_stable_milestone: int
+
+  @classmethod
+  def current(cls) -> Self:
+    """Computes the "current" QueryContext based on ambient information."""
```

Comment:
I'll have to use the phrase "ambient information" in code comments myself sometime.